noxfile.py

```diff
@@ -65,9 +65,10 @@ def format(session):
 
 @nox.session(python="3.12")
 def lint(session):
-    session.install("flake8", "black~=24.0", "isort")
+    session.install("flake8", "black~=24.0", "isort", "unasync", "setuptools")
     session.run("black", "--check", "--target-version=py37", *SOURCE_FILES)
     session.run("isort", "--check", *SOURCE_FILES)
+    session.run("python", "utils/run-unasync.py", "--check")
```

Comment:
Is this enough to make CI fail when the async/sync versions of a file are not synchronized?

Reply:
Yes. Here is an example output for the `lint` command after I added an `import re` in the _async version of document.py, without running unasync:

```
❯ nox -s lint
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python3.12 in .nox/lint
nox > python -m pip install flake8 'black~=24.0' isort unasync setuptools
nox > black --check --target-version=py37 setup.py noxfile.py docs/ elasticsearch_dsl/ examples/ tests/ utils/
All done! ✨ 🍰 ✨
91 files would be left unchanged.
nox > isort --check setup.py noxfile.py docs/ elasticsearch_dsl/ examples/ tests/ utils/
Skipped 1 files
nox > python utils/run-unasync.py --check
reformatted /Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/elasticsearch_dsl/_sync_check/search.py
reformatted /Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/elasticsearch_dsl/_sync_check/document.py
reformatted /Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/elasticsearch_dsl/_sync_check/index.py

All done! ✨ 🍰 ✨
3 files reformatted, 1 file left unchanged.
diff -x __pycache__ elasticsearch_dsl/_sync/document.py elasticsearch_dsl/_sync_check/document.py
18a19
> import re
Traceback (most recent call last):
  File "/Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/utils/run-unasync.py", line 101, in <module>
    main(check="--check" in sys.argv)
  File "/Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/utils/run-unasync.py", line 88, in main
    subprocess.check_call(
  File "/Users/miguelgrinberg/.pyenv/versions/3.12.0/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['diff', '-x', '__pycache__', 'elasticsearch_dsl/_sync', 'elasticsearch_dsl/_sync_check']' returned non-zero exit status 1.
nox > Command python utils/run-unasync.py --check failed with exit code 1
nox > Session lint failed.
```

Reply:
Thanks!